PurchasesCoreSwift/Public/EntitlementInfo.swift

```diff
@@ -0,0 +1,335 @@
+        self.willRenew = Self.willRenewWithExpirationDate(expirationDate: expirationDate,
+                                                          store: store,
+                                                          unsubscribeDetectedAt: unsubscribeDetectedAt,
+                                                          billingIssueDetectedAt: billingIssueDetectedAt)
+    }
+
+    private class func parseDate(dateString: String?, withDateFormatter dateFormatter: DateFormatter) -> Date? {
+        guard let dateString = dateString else {
+            return nil
+        }
+
+        return dateFormatter.date(from: dateString)
+    }
+
+    private class func isDateActive(expirationDate: Date?, forRequestDate requestDate: Date?) -> Bool {
+        guard let expirationDate = expirationDate else {
+            return true
+        }
+
+        let referenceDate: Date = requestDate ?? Date.init()
+        return expirationDate.timeIntervalSince(referenceDate) > 0
+    }
+
+    private class func parseOwnershipType(ownershipType: String?) -> PurchaseOwnershipType {
+        switch ownershipType {
+        case nil:
+            return .purchased
+        case "PURCHASED":
+            return .purchased
+        case "FAMILY_SHARED":
+            return .familyShared
+        default:
+            // TODO(post-migration check): Logging?
+            return .unknown
+        }
+    }
+
+    private class func parsePeriodType(periodType: String?) -> PeriodType {
+        switch periodType {
+        case "normal":
+            return .normal
+        case "intro":
+            return .intro
+        case "trial":
+            return .trial
+        default:
+            // TODO(post-migration check): Also handles nil.
+            return .normal
+        }
+    }
+
+    private class func parseStore(store: String?) -> Store {
+        switch store {
+        case "app_store":
+            return .appStore
+        case "mac_app_store":
+            return .macAppStore
+        case "play_store":
+            return .playStore
+        case "stripe":
+            return .stripe
+        case "promotional":
+            return .promotional
+        default:
+            // TODO(post-migration check): Logging?
+            return .unknownStore
+        }
+    }
+
+    private class func willRenewWithExpirationDate(expirationDate: Date?,
+                                                   store: Store,
+                                                   unsubscribeDetectedAt: Date?,
+                                                   billingIssueDetectedAt: Date?) -> Bool {
+        let isPromo = store == .promotional
+        let isLifetime = expirationDate == nil
+        let hasUnsubscribed = unsubscribeDetectedAt != nil
+        let hasBillingIssues = billingIssueDetectedAt != nil
+
+        return !(isPromo || isLifetime || hasUnsubscribed || hasBillingIssues)
+    }
+
+    public override var description: String {
```

Comment:
I usually try to put public stuff at the top, then internal, then private, so that most readers get to all the public stuff right away. 
For swift I usually make an exception for protocol conformance, which I do in an extension, to keep the protocol methods in one place. 

And for purely private stuff I usually end up moving it to a private extension as well for the sake of keeping the main declaration as clean as possible... with the obvious limitation being that stored properties have to be in the main declaration as well, so it ends up being a bit mixed. 

I don't think it's urgent, but perhaps we could come up with a consistent pattern and use that going forward for swift files?